### PR TITLE
Closes #1363 Use navbar_offcanvas_attributes set in .theme file.

### LINKED ISSUE #1363: Off canvas nav using demo ID and not using prescribed attribute variable from az_barrio.theme file in hook_preprocess_page
## Problem/Motivation

See https://github.com/az-digital/az_quickstart/blob/main/themes/custom/az_barrio/az_barrio.theme#L104-L108
There is an existing variable that is not being used `navbar_offcanvas_attributes` 

_Originally posted by @trackleft in https://github.com/az-digital/az_quickstart/issues/1353#issuecomment-1054630686_

### Describe the bug
The navbar off-canvas  menu is not using the prescribed attribute variable `navbar_offcanvas_attributes` and is instead recreating it, but missing the id.
https://github.com/az-digital/az_quickstart/blob/main/themes/custom/az_barrio/templates/layout/page.html.twig#L143

There are Google tag manager implications if we change the id, we will need to notify Marketing and Note this in our release notes.

### PR DESCRIPTION
**Pull request by trackleft**
Remove navbar off-canvas class logic from theme.
Use navbar_offcanvas_attributes var set in .theme file.
Add PHPUnit tests to check functionality and protect from regressions.

## Related Issue
Closes #1363 

## How to test this:
On install check that the appropriate classes and id exist.
By default, there is a block in the navbar region and a block in the navbar off_canvas region
When this is the case, there should be an element on the page that can be selected with this selector
`#navbar-top.has-navbar-region.has-navbar-offcanvas-region`

If you change the theme settings for arizona barrio to disable navbar off canvas there should no longer be an ID of `navbar-top` or any of these classes on any page using the arizona barrio theme.
```
has-navbar-region
has-navbar-offcanvas-region
no-navbar-region
no-navbar-offcanvas-region
```
if you re-enable navbar off canvas in the arizona barrio theme, the classes should reappear.

If you remove all blocks from the navbar region, you should now see an element that can be selected with this
`#navbar-top.no-navbar-region.has-navbar-offcanvas-region`

If you remove all blocks from the navbar off-canvas region, you should now see an element that can be selected with this
`#navbar-top.no-navbar-region.no-navbar-offcanvas-region`

If you add a block back to the navbar region, you should now see an element that can be selected with this
`#navbar-top.has-navbar-region.no-navbar-offcanvas-region`

If you add a block back to the navbar off-canvas region, you should now see an element that can be selected with this
`#navbar-top.has-navbar-region.has-navbar-offcanvas-region`

This is marked as `Breaking or visual change to existing behavior` since it could possibly have an effect on Google Tag Manager, or Google analytics tracking.

If you are tracking anything using the `navbarOffcanvasDemo` ID, you will need to change that to `navbar-top`
If you have custom CSS using the `navbarOffcanvasDemo` ID, you will need to change that to `navbar-top`
If you have custom JS using the `navbarOffcanvasDemo` ID, you will need to change that to `navbar-top`

## Types of changes

**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by smayden** [comment]
I'm noticing a visual difference with the search icon on the off-campus nav that I do not see on current live sites or other Probo sites. The search icon is a different color with a red outline and has active state on hover.

image

image

**Comment by joeparsons** [comment]
If this is still in progress should we convert it to draft?

**Comment by trackleft** [comment]
Oh, I thought that was done, I guess it is a draft now that I see Shoshana's comment.

**Comment by trackleft** [comment]
Fixed the search button issue.

File: .probo.yaml [hunk 1]
```diff
@@ -46,6 +46,7 @@ steps:
     plugin: Script
     script:
       - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush -y -n pm:enable az_demo
+      - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush -y -n pm:enable az_global_footer
```
Comment (trackleft) [inline .probo.yaml:49]:
```suggestion
```

**Review by joeparsons — APPROVED** [review]

**Comment by akslay** [comment]
It looks like the appearance of the desktop nav changed based on whether or not the off-canvas nav is enabled--is that intentional? (navbar looses its top/bottom borders when off-canvas is turned off).

**Comment by trackleft** [comment]
> It looks like the appearance of the desktop nav changes based on whether or not the off-canvas nav is enabled--is that intentional? (desktop navbar looses its top/bottom borders when off-canvas is turned off).

https://github.com/az-digital/az_quickstart/pull/1353#issuecomment-1054630686

**Comment by akslay** [comment]
> [#1353 (comment)](https://github.com/az-digital/az_quickstart/pull/1353#issuecomment-1054630686)

Ahh bug from another PR?

**Comment by trackleft** [comment]
@akslay I don't think it is supposed to lose the border if there is still a nav

**Review by akslay — APPROVED** [review]

**Comment by akslay** [comment]
Here's what the desktop nav looks like with the off-canvas-nav **enabled**:
Screen Shot 2022-05-12 at 3 05 31 PM


Here's what the desktop nav looks like with the off-canvas-nav **disabled**:
Screen Shot 2022-05-12 at 3 05 51 PM

**Comment by trackleft** [comment]
Oh, yeah, and you could just add the class to your theme settings too, forgot about that, thanks!

**Comment by joeparsons** [comment]
> Oh, yeah, and you could just add the class to your theme settings too, forgot about that, thanks!
> 
> In bootstrap the `navbar-light` class adds the border.

Is this a change that still needs to be made to this PR or is this done / ready to merge?

**Comment by trackleft** [comment]
> Is this a change that still needs to be made to this PR or is this done / ready to merge?

We don't need to change anything for this.

**Pull request merged**